components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,188 @@
+package softwareversionservice
+
+import (
+	"errors"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/getosutils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type ISoftwareVersionService interface {
+	GetSoftwareVersionServices(string) (models.SoftwareVersionDetails, error)
+}
+
+type SoftwareVersionService struct {
+	cmdCheckArray []string
+	osFilepath    string
+	logger        logger.Logger
+}
+
+func NewSoftwareVersionService(logger logger.Logger) ISoftwareVersionService {
+	return &SoftwareVersionService{
+		cmdCheckArray: cmdCheckArray,
+		osFilepath:    osFilepath,
+		logger:        logger,
+	}
+}
+
+const (
+	availability = " availability"
+	Ensure       = "Ensure "
+	osFilepath   = "/etc/os-release"
+)
+
+var cmdCheckArray = []string{"mkdir", "useradd", "chown", "rm", "touch", "truncate", "echo", "sleep", "ls", "grep", "yum", "which", "cp", "curl", "bash", "sysctl", "cat", "sed", "mount", "pvcreate", "vgcreate", "lvcreate", "mv", "systemd", "wget", "exec"}
+
+func (sv *SoftwareVersionService) GetSoftwareVersionServices(query string) (models.SoftwareVersionDetails, error) {
+	sv.logger.Debug("The query paramter entered = ", query)
+	serviceResponse := models.SoftwareVersionDetails{}
+	serviceResponse.Passed = true
+	serviceResponseArray := []models.Checks{}
+	cmdArray := []string{}
+	if query == "postgres" {
+		cmdArray = []string{"stat"}
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	} else if query == "opensearch" {
+		cmdArray = []string{"openssl"}
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	} else if query == "bastion" || query == "automate" || query == "chef-server" {
+		cmdArray = append(cmdArray, sv.cmdCheckArray...)
+	} else {
+		sv.logger.Error("The Query parameter is not supported")
+		return models.SoftwareVersionDetails{
+			Passed: false,
+			Checks: []models.Checks{
+				{},
+			},
+		}, errors.New("the query is not supported")
+	}
+
+	for i := 0; i < len(cmdArray); i++ {
+		checkResponse := sv.checkCommandVersion(cmdArray[i])
+		if !checkResponse.Passed {
+			serviceResponse.Passed = false
+		}
+		serviceResponseArray = append(serviceResponseArray, checkResponse)
+	}
+	osResponse, err := sv.checkOsVersion(sv.osFilepath)
+	if err != nil {
+		sv.logger.Error("Error while getting the OS Version = ", err)
+		return models.SoftwareVersionDetails{
+			Passed: false,
+			Checks: []models.Checks{
+				{},
+			},
+		}, err
```

Comment:
return the error properly along with custom message like errors.Wrap(err, "Error while getting the OS Version"). This error will be sent in response

Reply:
Done added the custom error message